toolchain/check/testdata/class/generic/adapt.carbon

```diff
@@ -301,7 +301,7 @@ fn ImportedConvertLocal(a: Adapter(C)) -> i32 {
 // CHECK:STDOUT:   %Main.import_ref.709: <witness> = import_ref Main//adapt_specific_type, loc10_1, loaded [concrete = constants.%complete_type.c07]
 // CHECK:STDOUT:   %Main.import_ref.feb = import_ref Main//adapt_specific_type, inst42 [no loc], unloaded
 // CHECK:STDOUT:   %Core.Int: %Int.type = import_ref Core//prelude/types/int, Int, loaded [concrete = constants.%Int.generic]
-// CHECK:STDOUT:   %.22b: %C.elem.66c = field_decl x, element0 [concrete]
+// CHECK:STDOUT:   %.22b: @C.%C.elem (%C.elem.66c) = field_decl x, element0 [concrete]
```

Comment:
I wonder if this change is testable? Something like:
```carbon
// --- a.carbon
library "a";
class X(T:! type) {
  var x: T;
}
// --- b.carbon
import library "a";
fn F(x: X(i32)) -> i32 {
  return x.x;
}
```
would maybe determine whether the type of the field `x` is being properly imported as attached.

Reply:
OK, I tried that, but the diff winds up being basically identical to this one (see commit 68f12ac23b). Is that what you expected?

Reply:
What I'd been expecting (or perhaps hoping?) is that a test of that form would fail to type-check prior to this change, because we'd import an unattached type for `X(T).x` and so the type of the `x.x` expression in `F` would be an unattached type `T` rather than being `i32`.

Reply:
Hm. It looks like examples like this do work currently, because we aren't directly referencing the `field_decl` -- we reference the `import_ref` instruction, which does have the right attached type. It's possible there's no way to observe this; if so, I think adding more tests covering the same thing probably isn't helpful.

Anyway, feel free to land this with or without the new test at your preference.